src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -607,21 +609,31 @@ public void run()
          previous = now;
 
          if (idleTimeout > 0L) {
-            logPoolState("Before cleanup\t");
             final List<PoolEntry> notInUseList = connectionBag.values(STATE_NOT_IN_USE);
             int removable = notInUseList.size() - config.getMinimumIdle();
-            for (PoolEntry poolEntry : notInUseList) {
-               if (removable <= 0) {
-                  break;
-               }
-               if (clockSource.elapsedMillis(poolEntry.lastAccessed, now) > idleTimeout) {
-                  if (connectionBag.reserve(poolEntry)) {
-                     closeConnection(poolEntry, "(connection passed idleTimeout)");
-                     removable--;
+            if (removable > 0) {
+               logPoolState("Before cleanup\t");
+               //sort pool entries on lastAccessed
+               Collections.sort(notInUseList, new Comparator<PoolEntry>() {
```

Comment:
sure. I thought you may not like pool entry implement default sort on last access.